### PR TITLE
finished 12 delete

### PR DESCRIPTION
**Pull request by scottkelly36**
unsure if i need more tests

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -7,390 +7,361 @@ const testData = require("../db/data/test-data/index");
...
+  test("400 invalid sort_by ", () => {
+    return request(app)
+      .get("/api/reviews?order=kk")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Invalid order");
+      });
+  });
+});
+describe.only("DELETE /api/comments/id", () => {
```
Comment (douglashellowell) [inline __tests__/app.test.js]: please delete `.only`s before commiting!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -7,390 +7,361 @@ const testData = require("../db/data/test-data/index");
...
+  test("return 204 when delete successful ", () => {
+    return request(app)
+      .delete("/api/comments/1")
+      .expect(204)
+      .then(({ body }) => {
+        expect(body).toEqual({});
+      });
```
Comment (douglashellowell) [inline __tests__/app.test.js]: a `204` cannot send a body back so there's no need to check here!

File: controllers/comments.controllers.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    .then(() => {
+      return removeComment(comment_id);
+    })
+    .then((result) => {
+      res.status(204).send({});
+    })
+    .catch((err) => {
+      next(err);
+    });
+};
```
Comment (douglashellowell) [inline controllers/comments.controllers.js]: some unusual syntax here

if you want to `checkExists` then continue after that promise has resolved then you should chain a `.then` off of `checkExists` itself

```diff
- Promise.resolve(checkId).then(/*...*/)
+ checkExists("comments", "comment_id", comment_id).then(/*...*/)
```

File: controllers/comments.controllers.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        });
+      } else {
+        return;
+      }
+    })
+    .then(() => {
+      return removeComment(comment_id);
+    })
+    .then((result) => {
+      res.status(204).send({});
```
Comment (douglashellowell) [inline controllers/comments.controllers.js]: you can use `res.sendStatus(204)` if you're not sending a body!

**Review by douglashellowell — CHANGES_REQUESTED** [review]
a refactor needed in your controller please!

**Review by Angaca — APPROVED** [review]
Nice, good to go!

**Pull request merged**